website/src/components/SearchPage/DownloadDialog/DownloadUrlGenerator.ts

```diff
@@ -0,0 +1,114 @@
+        const params = new URLSearchParams();
+
+        params.set('downloadAsFile', 'true');
+        params.set('downloadFileBasename', this.generateFilename(option.dataType));
+        if (!option.includeOldData) {
+            params.set(VERSION_STATUS_FIELD, versionStatuses.latestVersion);
+            params.set(IS_REVOCATION_FIELD, 'false');
+        }
+        if (!option.includeRestricted) {
+            params.set('dataUseTerms', 'OPEN');
+        }
+        if (option.dataType.type === 'metadata') {
+            params.set('dataFormat', metadataDefaultDownloadDataFormat);
+        }
+        if (option.compression !== undefined) {
+            params.set('compression', option.compression);
+        }
+
+        switch (downloadParameters.type) {
+            case 'filter':
+                const lapisSearchParameters = downloadParameters.lapisSearchParameters;
+                if (lapisSearchParameters.accession !== undefined) {
+                    for (const accession of lapisSearchParameters.accession) {
+                        params.append('accession', accession);
+                    }
+                }
+
+                const mutationKeys = [
+                    'nucleotideMutations',
+                    'aminoAcidMutations',
+                    'nucleotideInsertions',
+                    'aminoAcidInsertions',
+                ];
+
+                for (const [key, value] of Object.entries(lapisSearchParameters)) {
+                    // Skip accession and mutations
+                    if (key === 'accession' || mutationKeys.includes(key)) {
+                        continue;
+                    }
+                    const stringValue = String(value);
+                    const trimmedValue = stringValue.trim();
+                    if (trimmedValue.length > 0) {
+                        params.set(key, trimmedValue);
+                    }
+                }
+
+                mutationKeys.forEach((key) => {
+                    if (lapisSearchParameters[key] !== undefined) {
+                        params.set(key, lapisSearchParameters[key].join(','));
+                    }
+                });
+                break;
+            case 'select':
+                const sortedIds = Array.from(downloadParameters.selectedSequences).sort();
+                sortedIds.forEach((accessionVersion) => {
+                    params.append('accessionVersion', accessionVersion);
+                });
+                break;
+        }
+
+        return {
+            url: `${baseUrl}?${params}`,
+            baseUrl,
+            params,
+        };
+    }
+
+    private generateFilename(downloadDataType: DownloadDataType): string {
+        const siteName = kebabCase(this.websiteName);
+        const organism = kebabCase(this.organism);
+        const dataType = dataTypeForFilename(downloadDataType);
+        const timestamp = new Date().toISOString().slice(0, 16).replace('T', '').replace(':', '').replaceAll('-', '');
```

Comment:
IMO we should leave some separators in the datetime or it's not obvious it's a date-time and just seems like a very long weird number

Reply:
I have now opted for this format: `2024-10-22T1430` which is the ISO format with just the colon removed (which I think is not allowed in filenames on windows).

Reply:
I probably prefer T over `_` as we use `_` as the highest level separator already and date and time belong together, and aren't separate as for example filetype and organism.

I think most people will recognize the date either way, and the time probably too after they've done a few downloads.

So I'd go for `2024-10-22T1430`

Reply:
agree with Cornelius here, so I'll just leave it at that for now and as Theo said, we can also do this as a follow up change if we maybe gather more feedback and it's an issue.